blocks/api/raw-handling/comment-remover.js

```diff
@@ -8,5 +8,51 @@ export default function( node ) {
 		return;
 	}
 
+	if ( node.nodeValue.indexOf( 'more' ) === 0 ) {
+		// Grab any custom text in the comment
+		const matches = node.nodeValue.match( /more(.*)/ );
```

Comment:
this RegExp is pretty much asking for trouble. could we whitelist characters instead of blacklisting them?

```
<!--moredanghtmlcomments-->
```

Reply:
Thanks for having a look. I'm not sure what you mean by whitelisting, though. FWIW, this is how core does it:

https://github.com/WordPress/WordPress/blob/e50b9c943fddf781f980bba78d1ce927a5dc11b4/wp-includes/post.php#L532-L534

Reply:
Is this a valid more tag?

```
<!--moreYou should continue reading below!-->
```

Reply:
It is valid. :)

<img width="946" alt="screen shot 2018-02-15 at 18 17 25" src="https://user-images.githubusercontent.com/150562/36273463-83acb33a-127c-11e8-99ff-25cf1b890d8e.png">

Agreed on the greed, though.

Reply:
![](https://media3.giphy.com/media/vdpdWhpcIrtTi/200_d.gif)

Reply:
also, `/more(.*)/` is redundant. it matches the same things that `/more/` matches because both _nothing_ and _everything_ is `.*`

Reply:
> zero of everything and one of everything and many of everything is `.*`

![philosoraptor](https://user-images.githubusercontent.com/150562/36274100-8fdc23e6-127e-11e8-8f58-614eb42b9384.jpg)

>  `/more(.*)/` is redundant

Fair enough. We already check for `indexOf( 'more' ) === 0`, so why don't we just grab the substring starting after `'more'`? Then we can `.trim()` it. No RegExp needed.

Reply:
`nodeValue.slice(4).trim()`